router/router.go

```diff
@@ -1,40 +1,47 @@
 package router
 
 import (
-	"encoding/json"
 	"fmt"
 	"math"
 	"math/rand"
 	"sort"
-	"strconv"
 	"sync"
-	"sync/atomic"
 	"time"
 
 	"github.com/cenkalti/backoff/v4"
 	"github.com/rudderlabs/rudder-server/router/throttler"
 	"github.com/rudderlabs/rudder-server/services/diagnostics"
-	uuid "github.com/satori/go.uuid"
 
 	"github.com/rudderlabs/rudder-server/config"
 	"github.com/rudderlabs/rudder-server/jobsdb"
-	"github.com/rudderlabs/rudder-server/processor/integrations"
 	"github.com/rudderlabs/rudder-server/rruntime"
 	"github.com/rudderlabs/rudder-server/services/customdestinationmanager"
-	destinationdebugger "github.com/rudderlabs/rudder-server/services/destination-debugger"
 	"github.com/rudderlabs/rudder-server/services/stats"
 	"github.com/rudderlabs/rudder-server/utils/logger"
 	"github.com/rudderlabs/rudder-server/utils/misc"
 )
 
+type WorkerFactory interface {
+	CreateWorkers(rt *HandleT)
+	GetWorker(idx int) WorkerInterface
+	ResetSleep()
+}
+
+type WorkerInterface interface {
+	PostJobOnWorker(job *jobsdb.JobT)
+	WorkerProcess(rt *HandleT)
+	GetFailedJobIDMutex() *sync.RWMutex
```

Comment:
I was under the impression that failed job handling will only happen in final worker in pipeline and we shouldn't need this in batch worker.